### PR TITLE
Add event log call and new event.

### PR DESCRIPTION
**Pull request by robrap**
This is ready outside of tests.

### CODE REVIEW

**Comment by robrap** [comment]
FYI: @edx/arch-squad: I've got some questions in my code comments I want to discuss with someone.  Thanks.

File: package-lock.json [hunk 1]
```diff
@@ -9281,8 +9333,7 @@
         "ansi-regex": {
           "version": "2.1.1",
           "bundled": true,
-          "dev": true,
-          "optional": true
```
[the remark this answers is not in the document]
Comment (robrap) [reply package-lock.json]: I'm just going to ignore this and hope all is well.

File: src/services/NewRelicService.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  static logAPIErrorResponse(error) {
+    let { message } = error;
+    if (error.response) {
+      message = `${error.response.status} ${error.response.config.url} ${JSON.stringify(error.response.data)}`;
+    } else if (error.request) {
+      message = `${error.request.status} ${error.request.responseURL} ${error.request.responseText}`;
+    } else if (error.stack) {
+      message = error.stack;
+    }
+    this.logError(new Error(`API request failed: ${message}`));
```
Comment (abutterworth) [inline src/services/NewRelicService.js]: `this` is unneeded here?

File: src/services/NewRelicService.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+class NewRelicService {
```
[the remark this answers is not in the document]
Comment (abutterworth) [reply src/services/NewRelicService.js]: I think this is fine to leave in each project. If we do move it out I would suggest we change the name to something like LoggingService to abstract away New Relic. And then flesh out an API to do everything we need.

[the remark this answers is not in the document]
Comment (robrap) [reply src/services/NewRelicService.js]: Even here I could rename it.  I will do that.

File: src/services/NewRelicService.js (hunk 2 above)
Comment (robrap) [reply src/services/NewRelicService.js]: @abutterworth: Linter failed saying logError was undefined when I tried to remove this.

File: src/analytics.js [hunk 4]
```diff
@@ -1,11 +1,30 @@
...
+// Sends events to Segment and downstream
 const handleTrackEvents = (eventName, properties) => {
   // Simply forward track events to Segment
   window.analytics.track(eventName, properties);
 };
 
 
+// Sends events to tracking log and downstream
+// TODO: Determine consistent naming for eventName vs eventType and properties v eventData.
+const logEvent = (eventType, eventData) => {
```
Comment (abutterworth) [inline src/analytics.js]: Do we have an opinion on using this syntax vs below in this context?
```
function logEvent(eventType, eventData) {

}
```

Comment (robrap) [reply src/analytics.js]: I think you do, so I changed it. :)

Comment (abutterworth) [reply src/analytics.js]: Haha, I don't know if there's a proper way or not

Comment (robrap) [reply src/analytics.js]: The linter should state our proper way, and it is happy with both.  I like the functions more, and you seemed like you might, so I went for it.

Comment (robrap) [reply src/analytics.js]: @douglashall? @davidjoy?

**Pull request merged**